gradle-plugin/src/main/java/com/contrastsecurity/gradle/plugin/extensions/ContrastSDKService.java

```diff
@@ -0,0 +1,44 @@
+package com.contrastsecurity.gradle.plugin.extensions;
+
+import com.contrastsecurity.sdk.ContrastSDK;
+import com.contrastsecurity.sdk.UserAgentProduct;
+
+/** Creates and holds a static instance of the {@link ContrastSDK}. */
+public class ContrastSDKService {
+
+  public static ContrastSDK INSTANCE;
+
+  /**
+   * Initializes the SDK for the plugin. If configuration from the {@link
+   * ContrastConfigurationExtension} is missing or incorrect, provides a NOOP Instance
+   */
+  public static void initializeSdk(final ContrastConfigurationExtension extension) {
+    if (extension.getUsername() == null
```

Comment:
Once you change these props to `Property` you'll see you have new methods which will allow you avoid testing on null (basically the methods from `Optional`, e.g. `isPresent()`)